models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -164,7 +164,7 @@ public static function process(Asset $asset, Config $config, mixed $fileSystemPa
             $fileExtension = 'jpg';
         }
 
-        $filename .= '.' . $fileExtension;
+        $filename .= '.' . $config->getHash() . '.'. $fileExtension;
```

Comment:
Should we use a different separator for appending hashes? as dot `.` might cause issues with file extension detection.

Reply:
Why? File extension is the part after the last `.`.

Reply:
Agree with @jdreesen, `$ext = pathinfo($filename, PATHINFO_EXTENSION)` works great and I have never seen it confused by things like excessive punctuation.